airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/connector_runner.py

```diff
@@ -93,6 +95,7 @@ def call_read_with_state(self, config, catalog, state, **kwargs) -> List[Airbyte
         return output
 
     def run(self, cmd, config=None, state=None, catalog=None, raise_container_error: bool = True, **kwargs) -> Iterable[AirbyteMessage]:
```

Comment:
does the calling context automatically get this new config? if yes can you clarify how in a comment?

Reply:
All the tests using the `ConnectorRunner` are using the `connector_config` fixture (also function scoped) . The `connector_config` fixture  is loading the config from the `connector_config_path` fixture. The `connector_config_path` fixture dynamically retrieves the latest configuration written to the`updated_configurations` folder. So all tests using `connector_config` fixture are calling `docker_runner` methods with the "new" config.

I don't think clarifying the behavior here is a good idea as it's fixture related, not an internal logic of the runner. I wrote something [here](https://github.com/airbytehq/airbyte/blob/0cd81af095cf35229abc5e81feaa68629ea7a1e3/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/conftest.py#L59) in the `connector_config_path` fixture.